packages/suite/src/actions/wallet/fiatRatesActions.ts

```diff
@@ -64,10 +179,152 @@ export const handleRatesUpdate = () => async (dispatch: Dispatch, getState: GetS
+            return coinRates.lastWeek?.ts;
+        }
+        // no rates for localCurrency
+        return null;
+    };
+
+    const staleTickers = dispatch(getStaleTickers(lastWeekStaleFn, MAX_AGE_LAST_WEEK));
+
+    const promises = staleTickers.map(async ticker => {
+        try {
+            const response = await TrezorConnect.blockchainGetFiatRatesForTimestamps({
+                coin: ticker.symbol,
+                timestamps,
+            });
+            const results = response.success
+                ? response.payload
+                : await fetchLastWeekRates(ticker, localCurrency);
+
+            if (results?.tickers) {
+                dispatch({
+                    type: LAST_WEEK_RATES_UPDATE,
+                    payload: {
+                        symbol: ticker.symbol,
+                        tickers: results.tickers,
+                        ts: new Date().getTime(),
+                    },
+                });
+            }
+        } catch (error) {
+            console.log(error);
+        }
+    });
+    await Promise.all(promises);
+};
+
+/**
+ *  Fetch and update fiat rates for given `txs`
+ *  Primary source of rates is TrezorConnect, coingecko serves as a fallback
+ *
+ * @param {Account} account
+ * @param {AccountTransaction[]} txs
+ */
+export const updateTxsRates = (account: Account, txs: AccountTransaction[]) => async (
+    dispatch: Dispatch,
+) => {
+    if (txs?.length === 0 || isTestnet(account.symbol)) return;
+
+    const timestamps = txs.map(tx => tx.blockTime ?? getBlockbookSafeTime());
+    try {
+        const response = await TrezorConnect.blockchainGetFiatRatesForTimestamps({
+            coin: account.symbol,
+            timestamps,
+        });
+
+        const results = response.success
+            ? response.payload
+            : await getFiatRatesForTimestamps({ symbol: account.symbol }, timestamps);
+
+        if (results?.tickers) {
+            txs.forEach((tx, i) => {
+                dispatch({
+                    type: TX_FIAT_RATE_UPDATE,
+                    payload: {
+                        txid: tx.txid,
+                        updateObject: { rates: results.tickers[i]?.rates },
+                        account,
+                        ts: new Date().getTime(),
+                    },
+                });
+            });
+        }
+    } catch (error) {
+        console.error(error);
+    }
+};
+
+/**
+ * Fetch the account history (received, sent amounts, num of txs) for the given interval
+ * Returned data are grouped by month (weeks >= 52) or by day (weeks < 52)
+ * No XRP support
+ *
+ * @param {Account} account
+ * @param {number} weeks
+ * @returns
+ */
+export const fetchAccountHistory = async (account: Account, weeks: number) => {
+    const secondsInDay = 3600 * 24;
+    const secondsInMonth = secondsInDay * 30;
+
+    const startDate = subWeeks(new Date(), weeks);
+    const endDate = new Date();
+    try {
+        const response =
```

Comment:
TrezorConnect call should never fail with runtime error, this try/catch block looks like some kind of leftover